api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -92,6 +92,11 @@ message ExtAuthz {
   // or cannot be reached. The default status is HTTP 403 Forbidden.
   type.v3.HttpStatus status_on_error = 7;
 
+  // Sets the HTTP status that is returned to the client when the authorization server returns a
+  // response that is rejected by the filter / client. The default status is HTTP 500 Internal
+  // Server Error.
+  type.v3.HttpStatus status_on_rejected = 24;
```

Comment:
Does this need it's own config? Can't this just use `status_on_error`? I don't really see any difference between getting an error from the auth server and getting an invalid response.

Reply:
> @[tyxia](https://github.com/tyxia) tyxia [last week](https://github.com/envoyproxy/envoy/pull/33244#discussion_r1577055029) • 
> Rejecting the whole request is good, but I think we should make sure such rejection caused by invalid mutation is different from existing rejection caused by authorization failure. (e.g., the response code and responses body in localReply should be different)

> @[ggreenway](https://github.com/ggreenway) ggreenway [last week](https://github.com/envoyproxy/envoy/pull/33244#discussion_r1577072899)
> Agreed; the response code should be in the 5xx range, because this is a server-side bug in the auth service.

By default `status_on_error` is 403 (forbidden). I do think internal server error is a more appropriate status.

Reply:
Oh right, I remember that now. Yes, this makes sense. Although I'm still not sure it needs config; it could always be 500 (internal server error).

Reply:
Fine by me, I was just copying the pattern I saw for the error code